File: common/src/main/java/com/jon/common/presets/PresetDatabase.java
package com.jon.common.presets;

import androidx.room.Database;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

@Database(entities = {OutputPreset.class}, version = 3)
@TypeConverters({ ProtocolConverter.class })
public abstract class PresetDatabase extends RoomDatabase {
    static final String FILENAME = "presets.db";
    public abstract PresetDao presetDao();
}

File: common/src/main/java/com/jon/common/presets/PresetDatabase.kt
package com.jon.common.presets

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters

@Database(entities = [OutputPreset::class], version = 4)
@TypeConverters(ProtocolConverter::class)
abstract class PresetDatabase : RoomDatabase() {
    abstract fun presetDao(): IPresetDao

    companion object {
        const val FILENAME = "presets.db"
    }
}
